Notes/Algorithms/Data Structures/Hash-Tables.c

```diff
@@ -1,230 +1,244 @@
+// if key/value pair is not found return an ERROR code
+int searchHashTable(HashTable * hashTable, char key[], int value) {
+    int hashIndex = hashing(hashTable,key);
+    for (int i = 0; i < hashTable->size; i++) {
+        int index = (hashIndex + i) % hashTable->size;
+        if (hashTable->items[index]->value == value && strcmp(hashTable->items[index]->key, key) == 0) {
+            return SUCCESS;
+        }
     }
+    return ERROR;
 }
 
-int testSearch(int key, int expectedOutcome) {
-    //build the testHash
-    int keys[10] = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
-    int value[10]= { 10, 20, 30, 40, 50, 60, 70, 80, 90, 100};
-    buildHashTable( 9, keys, value);
-    int success = ERROR;
-    searchHashTable(9, key, &success);
+
+// testSearch
+// if the key/value pair is found successfully and the expectedOutcome matches a SUCCESS code 0 is returned then a success message is printed and a SUCCESS code is returned
+// if the value is not found then a failure message is printed and an ERROR -1 is returned
+// if the outcome does not match what is expected a mismatch message is printed and an ERROr -1 is returned 
+int testSearch(char key[], int value, int expectedOutcome) {
+    HashTable * testHash = createTestHashTable();
+    int success = searchHashTable(testHash, key, value);
     if (success == SUCCESS && expectedOutcome == SUCCESS) {
-        printf("The key %d was successfully found as expected. \n", key);
+        printf("The key/value pair: %s and %d was successfully found as expected. \n", key, value);
         return SUCCESS;
     } else if (success == ERROR && expectedOutcome == ERROR) {
-        printf("The key %d was not found as expected. \n", key);
+        printf("The key/value pair: %s and %d was not found as expected. \n", key, value);
         return SUCCESS;
     }
-    printf("Mismatch: An unknown error has occurred. \n");
+    printf("Mismatch has occurred. Result does not match expected outcome for success or failure. \n");
     return ERROR;
 }
 
+// buildSearchTests
+// creates 5 test cases expected to pass the testSearch function
+// creates 5 test cases (out of range) expected to fail the testSearch function
 void buildSearchTests() {
-printf("Build Search Test Passes: \n");
-    testSearch(1, SUCCESS);
-    testSearch(8, SUCCESS);
-    testSearch(3, SUCCESS);
-    testSearch(4, SUCCESS);
-    testSearch(9, SUCCESS);
+    printf("Build Search Test Passes: \n");
+    testSearch("KeyTwo", 20, SUCCESS);
+    testSearch("KeyEight", 80, SUCCESS);
+    testSearch("KeyThree", 30, SUCCESS);
+    testSearch("KeyFour", 40, SUCCESS);
+    testSearch("KeyNine", 90, SUCCESS);
 
     //5 Fail
     printf("Build Search Test Fails: \n");
-    testSearch(99, ERROR);
-    testSearch(-9, ERROR);
-    testSearch(11, ERROR);
-    testSearch(20, ERROR);
-    testSearch(-20, ERROR);
+    testSearch("KeyNinety", 0, ERROR);
+    testSearch("LaLa", 10, ERROR);
+    testSearch("Error", 20, ERROR);
```

Comment:
I might include an empty string as a test case, and maybe a really long string.

Reply:
Added empty string test case + long string case + checks to catch both before searching in testSearch
[commit 1178977aaed0264f58d48508b966b94b291a39ef](https://github.com/kcarrel/Training-Notes/pull/12/commits/1178977aaed0264f58d48508b966b94b291a39ef)